### ISSUE TITLE
[Jepsen][YSQL] Inconsistency in append test under partition

### ISSUE THREAD

**Issue by frozenspider** [yugabyte/yugabyte-db#7857]
During our Jepsen on Jenkins runs, we found an issue for `yb yugabyte-2.7.1.0-b66-centos-x86_64.tar.gz append nemesis partition`, which corresponds to YB @ acc0b7e241224cc6d8e5aa9606529a5829e1bb91 using yugabyte/jepsen@29471e91231e41500cf62e0345e8b0e8d0824dfa.

Complete log archive is too large to be attached in one go, so I've split it into relevant parts:
* [jepsen.log.zip](https://github.com/yugabyte/yugabyte-db/files/6228660/jepsen.log.zip)
* [history-and-graphs.zip](https://github.com/yugabyte/yugabyte-db/files/6228670/history-and-graphs.zip)
* [app-logs_n1-n2-n3.zip](https://github.com/yugabyte/yugabyte-db/files/6228672/app-logs_n1-n2-n3.zip)
* [app-logs_n4-n5.zip](https://github.com/yugabyte/yugabyte-db/files/6228677/app-logs_n4-n5.zip)

Possibly this is another symptom of #7729

**Comment by frozenspider** [comment]
Unlike #7729 , I was able to repro this on v2.0 and even v1.3.1.0!
E.g. using Jepsen's commit https://github.com/yugabyte/jepsen/commit/c6d3a65c64c8f3c28e44a2eb3af1a58b7aed6497 or https://github.com/yugabyte/jepsen/commit/d9e6a09a41db443a34d67c4233af9fe4da6ac5d5:
```
lein run test --os debian --version 1.3.1.0 -w ysql/append --nemesis partition --time-limit 200 --test-count 1 --concurrency 4n
```

**Comment by frozenspider** [comment]
Still reproducible after #7729 fix (YB @ c9e64df):
* [jepsen.log.zip](https://github.com/yugabyte/yugabyte-db/files/6400817/jepsen.log.zip)
* [history-and-graphs.zip](https://github.com/yugabyte/yugabyte-db/files/6400819/history-and-graphs.zip)
* [app-logs_n1-n2-n3.zip](https://github.com/yugabyte/yugabyte-db/files/6400822/app-logs_n1-n2-n3.zip)
* [app-logs_n4-n5.zip](https://github.com/yugabyte/yugabyte-db/files/6400824/app-logs_n4-n5.zip)

**Comment by frozenspider** [comment]
Newest log from 2.4.6.0-b3 (d1d2ab91c2ffa7f93b8542b6196a1384a1c72d3c) and yugabyte/jepsen@59499015cd04c9c7e351d8edfedf5c2380d8e1f6:
[2021-07-20_2.4_ysql_append_partition.zip](https://github.com/yugabyte/yugabyte-db/files/6849117/2021-07-20_2.4_ysql_append_partition.zip)

**Comment by frozenspider** [comment]
This was fixed in c9e64dfd95b0cc89d6592544703d75d4f3866a27 (#7729) and is no longer reproducible
